core/src/Kokkos_Tuners.hpp

```diff
@@ -315,17 +337,44 @@ class MultidimensionalSparseTuningProblem {
 
   using ValueArray = std::array<Kokkos::Tools::Experimental::VariableValue,
                                 space_dimensionality>;
+  template <class Key, class Value>
+  using extended_map = std::map<Key, Value>;
+  template <typename Key>
+  using extended_problem =
+      MultidimensionalSparseTuningProblem<extended_map, MaxDimensionSize, Key,
+                                          ProblemSpaceInput>;
+  template <typename Key, typename Value>
+  using ExtendedProblemSpace =
+      typename Impl::MapTypeConverter<extended_map<Key, Value>>::type;
+
+  template <typename Key>
+  auto extend(const std::string &axis_name,
+              const std::vector<Key> &new_tuning_axis) const
+      -> extended_problem<Key> {
+    ExtendedProblemSpace<Key, ProblemSpaceInput> extended_space;
+    for (auto &key : new_tuning_axis) {
+      extended_space.add_root_value(key);
+      extended_space.add_sub_container(m_space);
+    }
+    std::vector<std::string> extended_names = {axis_name};
+    for (const auto &name : extended_names) {
+      extended_names.push_back(name);
+    }
```

Comment:
You want to prepend `m_variable_names`, don't you?
```suggestion
    std::vector<std::string> extended_names;
    extended_names.reserve(m_variable_names.size()+1);
    extended_names.push_back(axis_name);
    extended_names.insert(extened_names.end(), 
                          m_variable_names.begin(), 
                          m_variable_names.end());
```